### PR TITLE
Adding release configs for lucene filtering

### PR DESCRIPTION
**Pull request by martin-gaievski**
Signed-off-by: Martin Gaievski <gaievski@amazon.com>

### Description
Adding release configs for Lucene filtering, it can be used to run perf tests for releases.
 
### Check List
- [X] Commits are signed as per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/k-NN/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by martin-gaievski** [comment]
Currently blocked, this PR should unblock k-NN main https://github.com/opensearch-project/k-NN/pull/661

**Comment by jmazanec15** [comment]
@martin-gaievski Can you rebase and try again? It should be fixed.

**Review by jmazanec15 — APPROVED** [review]

**Review by heemin32 — COMMENTED** [review]
Could you rename existing test file as well? We have index.json and text.yml in lucene-hnsw folder. Something like index-spec.json and default-test.yml?

File: benchmarks/perf-tool/release-configs/lucene-hnsw/relaxed-filter-test.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+test_name: "index-workflow"
+test_id: "Index workflow"
+num_runs: 10
+show_runs: false
+steps:
+  - name: delete_index
+    index_name: target_index
+  - name: create_index
+    index_name: target_index
+    index_spec: /home/ec2-user/[PATH]/index.json
```
Comment (heemin32) [inline benchmarks/perf-tool/release-configs/lucene-hnsw/relaxed-filter-test.yml:11]:
```suggestion
    index_spec: [INDEX_SPEC_PATH]/index.json
```

File: benchmarks/perf-tool/release-configs/lucene-hnsw/relaxed-filter-test.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  - name: ingest_multi_field
+    index_name: target_index
+    field_name: target_field
+    bulk_size: 500
+    dataset_format: hdf5
+    dataset_path: /home/ec2-user/data/sift-128-euclidean-with-attr.hdf5
```
Comment (heemin32) [inline benchmarks/perf-tool/release-configs/lucene-hnsw/relaxed-filter-test.yml:17]:
```suggestion
    dataset_path: [DATASET_PATH]/sift-128-euclidean-with-attr.hdf5
```

File: benchmarks/perf-tool/release-configs/lucene-hnsw/relaxed-filter-test.yml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  - name: refresh_index
+    index_name: target_index
+  - name: query_with_filter
+    k: 100
+    r: 1
+    calculate_recall: true
+    index_name: target_index
+    field_name: target_field
+    dataset_format: hdf5
+    dataset_path: /home/ec2-user/data/sift-128-euclidean-with-attr.hdf5
```
Comment (heemin32) [inline benchmarks/perf-tool/release-configs/lucene-hnsw/relaxed-filter-test.yml:29]:
```suggestion
    dataset_path: [DATASET_PATH]/sift-128-euclidean-with-attr.hdf5
```

File: benchmarks/perf-tool/release-configs/lucene-hnsw/relaxed-filter-test.yml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    neighbors_format: hdf5
+    neighbors_path: /home/ec2-user/data/sift-128-euclidean-with-filters.hdf5
```
Comment (heemin32) [inline benchmarks/perf-tool/release-configs/lucene-hnsw/relaxed-filter-test.yml:31]:
```suggestion
    neighbors_path: [DATASET_PATH]/sift-128-euclidean-with-filters.hdf5
```

File: benchmarks/perf-tool/release-configs/lucene-hnsw/relaxed-filter-test.yml [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    neighbors_dataset: neighbors_filter_5
+    filter_spec: /home/ec2-user/[PATH]/relaxed-filter-spec.json
```
Comment (heemin32) [inline benchmarks/perf-tool/release-configs/lucene-hnsw/relaxed-filter-test.yml:33]:
```suggestion
    filter_spec: [FILTER_SPEC_PATH]/relaxed-filter-spec.json
```

File: benchmarks/perf-tool/release-configs/lucene-hnsw/relaxed-filter-test.yml (hunk 1 above)
Comment (heemin32) [reply benchmarks/perf-tool/release-configs/lucene-hnsw/relaxed-filter-test.yml:11]: I think `/home/ec2-user` shouldn't be specified here.

**Comment by martin-gaievski** [comment]
> Could you rename existing test file as well? We have index.json and text.yml in lucene-hnsw folder. Something like index-spec.json and default-test.yml?

I think it's better to create new nested folder(s) for filtering. Renaming existing test config file may cause confusion/errors as its not consistent with tests for other engines.

**Comment by heemin32** [comment]
> > Could you rename existing test file as well? We have index.json and text.yml in lucene-hnsw folder. Something like index-spec.json and default-test.yml?
> 
> I think it's better to create new nested folder(s) for filtering. Renaming existing test config file may cause confusion/errors as its not consistent with tests for other engines.

We can change the name in other test cases as well. The reason that I think renaming might be better is because index-spec.json file is used for filtering test as well.

File: benchmarks/perf-tool/release-configs/lucene-hnsw/relaxed-filter-test.yml (hunk 1 above)
Comment (martin-gaievski) [reply benchmarks/perf-tool/release-configs/lucene-hnsw/relaxed-filter-test.yml]: ack

File: benchmarks/perf-tool/release-configs/lucene-hnsw/relaxed-filter-test.yml (hunk 2 above)
Comment (martin-gaievski) [reply benchmarks/perf-tool/release-configs/lucene-hnsw/relaxed-filter-test.yml]: ack

File: benchmarks/perf-tool/release-configs/lucene-hnsw/relaxed-filter-test.yml (hunk 3 above)
Comment (martin-gaievski) [reply benchmarks/perf-tool/release-configs/lucene-hnsw/relaxed-filter-test.yml]: ack

File: benchmarks/perf-tool/release-configs/lucene-hnsw/relaxed-filter-test.yml (hunk 4 above)
Comment (martin-gaievski) [reply benchmarks/perf-tool/release-configs/lucene-hnsw/relaxed-filter-test.yml]: ack

File: benchmarks/perf-tool/release-configs/lucene-hnsw/relaxed-filter-test.yml (hunk 5 above)
Comment (martin-gaievski) [reply benchmarks/perf-tool/release-configs/lucene-hnsw/relaxed-filter-test.yml]: ack

**Pull request merged**